internal/db/db.go

```diff
@@ -117,3 +186,31 @@ func (db *dB) GetComposes(accountNumber string, limit, offset int) ([]ComposeEnt
 
 	return composes, count, nil
 }
+
+func (db *dB) CountComposesFor(accountNumber string, timeIntervals ...TimeInterval) (int, error) {
+	ctx := context.Background()
+	conn, err := db.Pool.Acquire(ctx)
+	if err != nil {
+		return 0, err
+	}
+	defer conn.Release()
+
+	intervals := ""
+	for _, timeInterval := range timeIntervals {
+		sTimeInterval, err := timeInterval.ToString()
+		if err != nil {
+			return 0, err
+		}
+		intervals = fmt.Sprintf("%s %s", intervals, sTimeInterval)
+	}
+
+	var count int
+	err = conn.QueryRow(ctx, fmt.Sprintf(
+		"SELECT COUNT(*) FROM composes WHERE account_number=$1 AND created_at > CURRENT_TIMESTAMP%s", intervals),
+		accountNumber).Scan(&count)
```

Comment:
Don't use `fmt.Sprintf` in queries, rely on the formatting and sanitizing done by pgx.

And with `time.Duration`: `SELECT COUNT(*) FROM composes WHERE account_number=$1 AND CURRENT_TIMESTAMP - created_at <=  $2` works with `time.Duration` for $2.